scripts/irods/test/test_igroupadmin.py

```diff
@@ -142,6 +142,26 @@ def test_group_administrator_scenarios__issue_6888(self):
             self.admin.assert_icommand(['iadmin', 'rmuser', rodsuser_name])
             self.admin.assert_icommand(['iadmin', 'rmuser', self.groupadmin2.username])
 
+    def test_groupadmin_can_lock_self_out_of_group__issue_7707(self):
+        """A groupadmin should be able to add a regular user to an empty group, thereby losing the ability to manage the group."""
+        test_username = 'bob'
+        try:
+            self.groupadmin.assert_icommand(['igroupadmin', 'mkgroup', self.group])
+            self.groupadmin.assert_icommand(['igroupadmin', 'mkuser', test_username])
+
+            # Populate empty group. groupadmin should no longer have any control of this group now
+            self.groupadmin.assert_icommand(['igroupadmin', 'atg', self.group, test_username])
+
+            # Assert groupadmin cannot control this group now
+            self.groupadmin.assert_icommand_fail(['igroupadmin', 'atg', self.group, self.groupadmin.username])
+            self.groupadmin.assert_icommand_fail(['igroupadmin', 'rfg', self.group, test_username])
+            self.groupadmin.assert_icommand_fail(['igroupadmin', 'lg', self.group])
```

Comment:
What iRODS error codes do these commands print out?
I'm thinking we should assert the iRODS error codes too. Thoughts?

Reply:
That's fair. Hopefully something along the lines of `CAT_NO_ACCESS_PERMISSION`...

Reply:
Done.